components/nexus-core/src/main/java/org/sonatype/nexus/util/file/DirSupport.java

```diff
@@ -402,7 +434,11 @@ public static boolean copyIfExists(final Path from, final Path to, final @Nullab
   public static void move(final Path from, final Path to)
       throws IOException
   {
-    Files.move(from, to, StandardCopyOption.REPLACE_EXISTING);
+    if (areOnSameFileStore(from, to)) {
+      sameFileStoreMove(from, to);
```

Comment:
Maybe consider falling back to `crossFileStoreMove` if this throws an `IOException`?  I believe the code used to do that before it was refactored, but I might be mis-remembering.